src/Analyser/TypeSpecifier.php

```diff
@@ -1114,11 +1116,19 @@ private function specifyTypesForCountFuncCall(
 				$valueTypesBuilder = ConstantArrayTypeBuilder::createEmpty();
 				for ($i = 0; $i < $sizeType->getMin(); $i++) {
 					$offsetType = new ConstantIntegerType($i);
+					$hasOffset = $arrayType->hasOffsetValueType($offsetType);
+					if ($hasOffset->no()) {
+						break;
+					}
 					$valueTypesBuilder->setOffsetValueType($offsetType, $arrayType->getOffsetValueType($offsetType));
 				}
 				if ($sizeType->getMax() !== null) {
 					for ($i = $sizeType->getMin(); $i < $sizeType->getMax(); $i++) {
 						$offsetType = new ConstantIntegerType($i);
+						$hasOffset = $arrayType->hasOffsetValueType($offsetType);
+						if ($hasOffset->no()) {
+							break;
+						}
```

Comment:
realized that one of the new tests started creating constant arrays with dozens of `*ERROR*` values, so I consistently added these guard checks.

Reply:
would it make sense to return null when errors start popping up, instead of returning everything until the first error?

Reply:
simplified. WDYT?